clang/include/clang/Basic/Attr.td

```diff
@@ -1313,6 +1313,40 @@ def SYCLIntelNoGlobalWorkOffset : InheritableAttr {
   let PragmaAttributeSupport = 0;
 }
 
+def SYCLIntelLoopFuse : InheritableAttr {
+  let Spellings = [CXX11<"intel","loop_fuse">];
+  let Args = [ExprArgument<"Value", /*optional=*/ 1>];
+  let LangOpts = [SYCLIsDevice, SYCLIsHost];
+  let Subjects = SubjectList<[Function], ErrorDiag>;
+  let Documentation = [SYCLIntelLoopFuseDocs];
+  let PragmaAttributeSupport = 0;
+  let AdditionalMembers = [{
+    static unsigned getMinValue() {
+      return 0;
+    }
+    static unsigned getMaxValue() {
+      return 1024*1024;
+    }
+  }];
+}
+
+def SYCLIntelLoopFuseIndependent : InheritableAttr {
```

Comment:
@AaronBallman I took a look at this. I could replace it with one definition but I'm not sure if that's the right thing to do since the 2 'spellings' would then have to be treated as conflicting attributes. Theoretically yes, I can throw a 'conflicting attribute' diagnostic by using accessor to distinguishing between the 2 spellings, but is that correct practice?

Reply:
I don't think we really have an established practice here.

Usually, when two attributes are mutually exclusive, it's because they do opposite things in the same space and so sharing a semantic attribute would lead to more confusing code (such as the `hot` and `cold` attributes).

In this case, it seems like the two attributes are doing the same thing but with slight variation between them. If that's the case, then having a single semantic attribute allows you to express the semantics of the attribute a bit more directly.

I definitely don't insist on a change here if you think it's more clear to use two semantic attributes.

Reply:
I agree with you :) I think one semantic attribute makes sense here. So I changed it.